File: core/src/main/java/core/rag/Question.java
package core.rag;

public interface Question {
    String enrollQuestion(String userId, String title, String category, String contents);
    String updateQuestion(String userId, String questionId, String category, String contents);
    String deleteQuestion(String userId, String questionId);
    QuestionDTO readQuestion(String questionId);
    String readQuestionTitles(String category, int startNum, int endNum);
}

File: core/src/main/kotlin/core/rag/Question.kt
package core.rag

import java.time.LocalDateTime

data class Question(
    val category: String,
    val title: String,
    val content: String,
    val userId: String,
    val questionId: String? = null,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null
)